DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetAccountLoginWindow.cs

```diff
@@ -0,0 +1,183 @@
+            lblPlayerEmail.Text = "Email:";
+            lblPlayerEmail.ClientRectangle = new Rectangle(12, tbPlayerEmail.ClientRectangle.Y + 1,
+                lblPlayerEmail.ClientRectangle.Width, lblPlayerEmail.ClientRectangle.Height);
+
+            lblPlayerPassword = new XNALabel(WindowManager);
+            lblPlayerPassword.Name = "lblPlayerPassword";
+            lblPlayerPassword.FontIndex = 1;
+            lblPlayerPassword.Text = "Password:";
+            lblPlayerPassword.ClientRectangle = new Rectangle(
+                12, tbPlayerEmail.ClientRectangle.Y + 35,
+                lblPlayerPassword.ClientRectangle.Width,
+                lblPlayerPassword.ClientRectangle.Height
+            );
+
+            tbPlayerPassword = new XNAPasswordBox(WindowManager);
+            tbPlayerPassword.Name = "tbPlayerPassword";
+            tbPlayerPassword.Text = "Password";
+            tbPlayerPassword.ClientRectangle = new Rectangle(100, lblPlayerPassword.ClientRectangle.Y, 200, 19);
+            tbPlayerPassword.Text = "";
+
+            lblError = new XNALabel(WindowManager);
+            lblError.Name = "lblPlayerPassword";
+            lblError.FontIndex = 1;
+            lblError.TextColor = Color.Red;
+            lblError.Text = "";
+            lblError.ClientRectangle = new Rectangle(
+                12, tbPlayerPassword.ClientRectangle.Y + 35,
+                lblError.ClientRectangle.Width,
+                lblError.ClientRectangle.Height
+            );
+
+            AddChild(tbPlayerEmail);
+            AddChild(tbPlayerPassword);
+            AddChild(lblPlayerEmail);
+            AddChild(lblPlayerPassword);
+            AddChild(btnLogin);
+            AddChild(btnCancel);
+            AddChild(btnRegister);
+            AddChild(lblError);
+
+            base.Initialize();
+
+            CenterOnParent();
+
+            Keyboard.OnKeyPressed += Keyboard_OnKeyPressed;
+        }
+
+        private void BtnRegister_LeftClick(object sender, EventArgs e)
+        {
+            Process.Start(CnCNetAPI.API_REGISTER_URL);
+        }
+
+        private void Keyboard_OnKeyPressed(object sender, KeyPressEventArgs e)
+        {
+            if (Enabled)
+            {
+                switch (e.PressedKey)
+                {
+                    case Keys.Enter:
+                        BtnLogin_LeftClick(this, EventArgs.Empty);
+                        break;
+
+                    case Keys.Tab:
+                        if (tbPlayerEmail.IsActive)
+                        {
+                            WindowManager.SelectedControl = tbPlayerPassword;
+                        }
+                        else
+                        {
+                            WindowManager.SelectedControl = tbPlayerEmail;
+                        }
+                        break;
+                }
+            }
+        }
+
+        private void BtnCancel_LeftClick(object sender, EventArgs e)
+        {
+            Cancel?.Invoke(this);
```

Comment:
When `Cancel` is of type `EventHandler` instead of `Action<object>`, pass `EventArgs.Empty` to it as a second parameter.